children/tests/test_api.py

```diff
@@ -1235,6 +1242,148 @@ def test_children_offset_pagination(
+    has been over 20. Then the pages after 2nd page all failed.
+    """
+    # There should be tens of items of data and the last page should not be full.
+    total_count = 70
+    limit = 20  # There should be more than 2 pages.
+    offset = (page - 1) * limit
+
+    ChildWithGuardianFactory.create_batch(total_count, project=project)
+
+    variables = {"projectId": get_global_id(project), "limit": limit, "offset": offset}
+    executed = project_user_api_client.execute(
+        CHILDREN_PAGINATION_QUERY,
+        variables=variables,
+    )
+
+    assert len(executed["data"]["children"]["edges"]) == expected_item_count_in_page
+
+
+@pytest.mark.parametrize("page", list(range(1, 5)))
+@pytest.mark.parametrize("limit", [2, 10, 20, 25])
+def test_children_pagination_cursor_works_the_same_with_offset_and_after(
+    project_user_api_client, project, page, limit
+):
+    """
+    Check that the query returns same results no matter which one
+    is used with the pagination,the combination of a limit and an offset
+    or a first and an after.
+    NOTE: There has been problems when there has been a greater amount of data
+    (~70 items in KK-1049) and page size has been over 20.
+    Then the pages after 2nd page all failed.
+    """
+    query = """
+    query Children($projectId: ID!, $limit: Int, $offset: Int, $after: String, $first: Int) {
+        children(projectId: $projectId, limit: $limit, offset: $offset, after: $after, first: $first) {
+            count
+            edges {
+            cursor
+            node {
+                lastName
+            }
+            }
+        }
+    }
+    """  # noqa: E501,
+
+    ChildWithGuardianFactory.create_batch(110, project=project)
+
+    offset = (page - 1) * limit
+    executed_with_offset = project_user_api_client.execute(
+        query,
+        variables={
+            "projectId": get_global_id(project),
+            "limit": limit,
+            "offset": offset,
+        },
+    )
+
+    # NOTE: There will be 1 result less in the after-query,
+    # because the after-parameter is read from the first result
+    # and the cursor set in after, is not included in the result set.
+    first = limit - 1
+    after = executed_with_offset["data"]["children"]["edges"][0]["cursor"]
+    executed_with_after = project_user_api_client.execute(
+        query,
+        variables={
+            "projectId": get_global_id(project),
+            "first": first,
+            "after": after,
+        },
+    )
+
+    assert len(executed_with_after["data"]["children"]["edges"]) == first
+    assert (
+        executed_with_after["data"]["children"]["edges"]
+        == executed_with_offset["data"]["children"]["edges"][1:]
+    )
+
+
+@pytest.mark.parametrize("page", list(range(1, 5)))
+@pytest.mark.parametrize("limit", [1, 2, 10, 20])
+def test_children_pagination_cursor_generation(
+    project_user_api_client, project, page, limit
+):
+    """
+    Test that the cursors are created as invented.
```

Comment:
What do you mean by "as invented"? Something similar to "as expected" or something else? Maybe the wording could be made clearer here although the intent can be guessed.